### PR TITLE
added nocam2map tests

### PR DESCRIPTION
**Pull request by Kelvinrr**
## Description

Nocam2map translated and tested. 

## Motivation and Context

451Mb

## How Has This Been Tested?

old and new tests pass locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/FunctionalTestsNoCam2Map.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  LineManager latline(*latCube);
+  double latValue = -60;
+  for(latline.begin(); !latline.end(); latline++) {
+    for(int i = 0; i < latline.size(); i++) {
+      latValue += 0.0001;
+      latline[i] = (double) latValue ;
+    }
+    latCube->write(latline);
+  }
+  latCube->close();
```
Comment (Kelvinrr) [inline isis/tests/FunctionalTestsNoCam2Map.cpp:35]: This is redundant between the two tests, can't really justify factoring it out as I cannot imagine other tests need Lat/Lon cubes. I'll double check in the AM and if there is a good reason to make it a fixture I can,

File: isis/src/base/apps/nocam2map/main.cpp [hunk 2]
```diff
@@ -48,738 +48,30 @@ map <QString, void *> GuiHelpers() {
...
-  UserInterface &ui = Application::GetUserInterface();
-
-  // Get mapping group from map file
-  Pvl userMap;
-  userMap.read(ui.GetFileName("MAP"));
-  PvlGroup &userGrp = userMap.findGroup("Mapping", Pvl::Traverse);
-
-  //Write map file out to the log
-  Isis::Application::GuiLog(userGrp);
-}
```
Comment (acpaquette) [inline isis/src/base/apps/nocam2map/main.cpp:782]: Any particular reason this got moved?

File: isis/src/base/apps/nocam2map/nocam2map.h [hunk 3]
```diff
@@ -1,10 +1,13 @@
-#ifndef nocam2map_h
-#define nocam2map_h
+#ifndef NoCam2Map_h
+#define NoCam2Map_h
```
Comment (acpaquette) [inline isis/src/base/apps/nocam2map/nocam2map.h]: We have made most of the header defines all lowercase, may want to continue following that trend

File: isis/tests/FunctionalTestsNoCam2Map.cpp (hunk 1 above)
Comment (acpaquette) [reply isis/tests/FunctionalTestsNoCam2Map.cpp:35]: Could make it into a fixture local to these tests, but I'm fine with it either way

**Review by acpaquette — COMMENTED** [review]
Looks good! It does look like there are a lot of untested features/gui parameters in this app that the old tests don't do a good job covering. I don't know how much we care, especially since I have never even heard of nocam2map nor used it.

File: isis/src/base/apps/nocam2map/main.cpp (hunk 2 above)
Comment (Kelvinrr) [reply isis/src/base/apps/nocam2map/main.cpp:782]: Gui Helpers got shuffled when moved back into Main.cpp but it's still there. https://github.com/USGS-Astrogeology/ISIS3/blob/ef3f93470d3554c717c8b266fab07320a281e649/isis/src/base/apps/nocam2map/main.cpp#L118

**Comment by Kelvinrr** [comment]
@acpaquette Yeah, in doing these tests I was concerned that GUIHelpers are not being tested. But this is technically true about the old tests so 🤷🏽

**Comment by acpaquette** [comment]
Mhm, do we want to try to add the extra tests here or are we cool with just merging the converted tests?

**Comment by kaitlyndlee** [comment]
I think we want to open an issue about the original tests not covering everything and move on.

**Comment by Kelvinrr** [comment]
I dont think it's even possible to test without a GUI testing suite, these are functions that are only called in GUI mode and that's another can of worms on it's own.

**Review by kaitlyndlee — APPROVED** [review]

**Pull request merged**